src/mapbox/extended_nav_control.js

```diff
@@ -38,7 +39,7 @@ export default class ExtendedControl {
       'direction',
       () => {
         Telemetry.add(Telemetry.MAP_ITINERARY);
-        window.app.navigateTo('/routes');
+        fire('geolocateControlClick');
```

Comment:
As a consequence, the button has no effect when it's used on the ServicePanel.

As @bbecquet had suggested, I think it would be cleaner to fire the event this other way around, to send a `poi` value here.